### ISSUE TITLE
Javadoc formatting escapes '{@code' sometimes

### ISSUE THREAD

**Issue by ViToni** [eclipse-jdt/eclipse.jdt.core#37]
Tested with  Eclipse 2022-03 and default formatting settings (and also custome ones)
When trying to use formatting the '@' in '{@code' is escaped sometimes

```Java
/**
 * <h2>Works</h2>
 * 
 * <pre>
 * {@code
final Object o = ...
 * }
 * </pre>
 *
 * <pre>
 * {@code
for (int i = 0; i< 4; i++) {
    foo();
}
 * }
 * </pre>
 *
 * <h2>Doesn't work</h2>
 * 
 * <pre>
 * {@code
final Object o = new Object();
 * }
 * </pre>
 *
 * <pre>
 * {@code
final int i = 4;
 * }
 * </pre>
 *
 * <pre>
 * {@code
class BaseImpl
        implements A, B, C {
}
class SubImpl
        extends BaseImpl
        implements X, Y, Z {
}
 * }
 * </pre>
 *
 */
public interface FormatTest {
}
```
the result is:
```Java
/**
 * <h2>Works</h2>
 * 
 * <pre>
 * {@code
final Object o = ...
 * }
 * </pre>
 *
 * <pre>
 * {@code
for (int i = 0; i< 4; i++) {
    foo();
}
 * }
 * </pre>
 *
 * <h2>Doesn't work</h2>
 * 
 * <pre>
 * {
 * 	&#64;code
 * 	final Object o = new Object();
 * }
 * </pre>
 *
 * <pre>
 * {
 * 	&#64;code
 * 	final int i = 4;
 * }
 * </pre>
 *
 * <pre>
 * {
 * 	&#64;code
 * 	class BaseImpl implements A, B, C {
 * 	}
 * 	class SubImpl extends BaseImpl implements X, Y, Z {
 * 	}
 * }
 * </pre>
 *
 */
public interface FormatTest {
}
```
Trying to create Javadoc out of the original file works without issues with `javadoc` (tested with `javadoc` from Java 8 & Java 17)

**Comment by jjohnstn** [comment]
Hello, while I was able to reproduce the problem with the default Eclipse built-in format profile in 2022-03, the problem does not exist in 4.24 M1.  If you go to: https://download.eclipse.org/eclipse/downloads/drops4/S-4.24M1-202204070240/

you can download for your OS and verify.  I am closing this as fixed.

**Comment by ViToni** [comment]
Confirmed as resolved with 20220414-1707 (that's the version the Eclipse installer installed on its own as latest 2022-06),

**Comment by jjohnstn** [comment]
Great.
